single_kernel_mongo/state/charm_state.py

```diff
@@ -457,6 +458,21 @@ def set_scaling_down(self, rel_id: int, departing_unit_name: str) -> bool:
         self.unit_peer_data.update({rel_departed_key: json.dumps(scaling_down)})
         return scaling_down
 
+    def share_status_with_config_server(self, status: StatusBase):
+        """Shares this shard's status with the config server."""
+        if not self.is_role(MongoDBRoles.SHARD):
+            return
+        if not self.shard_relation:
```

Comment:
should this be `config_relation`?

Reply:
This is probably poorly named but `shard_relation` is the relation on the shard side (`sharding` is the relation name)

Reply:
yeah I believe this should be renamed (since shards are related to config servers, and config-servers are related to shards)

Reply:
If you don't mind, I'll create a ticket to do that after all is merged otherwise this will be hell to rename and will create so many conflicts: https://warthogs.atlassian.net/browse/DPE-6321